packages/core/lib/events/DomainEventEmitter.ts

```diff
@@ -162,10 +173,10 @@ export class DomainEventEmitter<SupportedEvents extends CommonEventDefinition[]>
       await this.executeEventHandler(event, handler, false)
     }
 
-    const bgPromises = eventHandlers.background.map((handler) =>
-      this.executeEventHandler(event, handler, true),
-    )
-    Promise.all(bgPromises).then(() => {
+    const bgPromise = Promise.all(
+      eventHandlers.background.map((handler) => this.executeEventHandler(event, handler, true)),
+    ).then(() => {
+      this.inProgressBackgroundHandlerByEventId.delete(event.id)
```

Comment:
this potentially leaks memory, because promises won't be removed until we call emitter dispose.

Can we add `then` block to every inProgress handler that would remove them from the map, when initiating the process?

Reply:
wait, no, I misread the code. this is not invoked on disposal, but on processing, so all good.